src/index.js

```diff
@@ -1,37 +1,66 @@
 const { SQSClient, SendMessageCommand } = require('@aws-sdk/client-sqs')
-const { logger, serializeError } = require('./logging')
+const { SNSClient, PublishCommand } = require('@aws-sdk/client-sns')
+const { logger: defaultLogger, serializeError } = require('./logging')
+const config = require('./config')
+const { createS3URL } = require('./s3')
+const { createClientConfig } = require('./aws')
 const {
   SQS_INDEXER_QUEUE_URL: indexerQueue,
   SQS_INDEXER_QUEUE_REGION: indexerQueueRegion
 } = process.env
 
-const SQSclient = new SQSClient({
+const defaultSQSClient = new SQSClient({
   region: indexerQueueRegion
 })
 
 /**
- * @type {import('aws-lambda').SNSHandler}
+ * Create the main function that will handle SNS events
+ * @param {object} arg
+ * @param {string} arg.eventsTopic - events pubsub topic - e.g. an AWS SNS TopicArn
+ * @param {import('@aws-sdk/client-sns').SNSClient} [arg.eventsTopicClient] - client to use to send commands to eventsTopic
+ * @returns {import('aws-lambda').SNSHandler}
  */
-async function main(event) {
-  try {
-    const records = toS3Event(event).Records
-      .filter(r => r.eventName.startsWith('ObjectCreated'))
-      .filter(r => r.s3.object.key.endsWith('.car'))
-    for (const record of records) {
-      const snsMessage = `${record.awsRegion}/${record.s3.bucket.name}/${record.s3.object.key}`
-      await publishToSQS(snsMessage)
+function createBucketToIndexerLambda({
+  eventsTopic = config.eventsTopic,
+  eventsTopicClient = new SNSClient(createClientConfig({ topic: config.eventsTopic })),
```

Comment:
The main difference from https://github.com/elastic-ipfs/bucket-to-indexer/pull/11 is that this client is specific to `eventsTopic` and the default value has the region set to a value parsed from the `eventsTopic` ARN (if it's a parseable ARN)

Reply:
@francardoso93 I think this makes sense compared to having multiple environment variables, because doing it this way can't result in a situation where the two env vars don't work together.

> We could use the same value as indexerQueueRegion for the SNS client (Or other env if we want to decouple).

I don't think this is ideal because it doesn't seem like it would easily model a situation where the indexerQueue and eventsTopic are in different regions. (not the case now, but it may be desireable in a future context and it would be nice not to have to adjust the lambda code for that situation)

Reply:
I think it's a good idea to pick the region for the client out of the configured topic ARN...for the reasons @gobengo mentions.

I'm not normally one to take on complexity but I think in this case it's worth it since it can mitigate developer error and can be fully tested.

I appreciate this is not consistent with the config for the indexer queue. If the region for the SQS queue client can be extracted from the configured queue URL and they always match then I'd suggest opening an issue to refactor that also.